src/connectivity.jl

```diff
@@ -518,19 +518,20 @@ function _neighborhood(g::AbstractGraph{T}, v::Integer, d::Real, distmx::Abstrac
         seen[src] && continue
         seen[src] = true
         currdist = dists[src]
+        push!(neighs, src)
         vertexneighbors = neighborfn(g, src)
         @inbounds for vertex in vertexneighbors
-            if !seen[vertex] 
+            if !seen[vertex]  && currdist < typemax(U)
                 vdist = currdist + distmx[src, vertex]
-                if vdist <= d
+                if vdist <= ud  && vdist < dists[vertex]
                     push!(Q, vertex)
-                    push!(neighs, vertex)
                     dists[vertex] = vdist
                 end
             end
         end
     end
-    return [(x::T, dists[x]::U) for x in neighs]
+    # working around closure bug - should be fixed in Julia 1.2
```

Comment:
Can you add a link the bug issue so that our future selves will know when to remove this workaround?